### ISSUE TITLE
Gnerative Adversarial Networks

### ISSUE THREAD

**Issue by prathimacode-hub** [prathimacode-hub/DS-ScriptsNook#101]
Hey contributors, good to see you here

This issue will helps readers in giving all the guidance that one needs to learn about Gnerative Adversarial Networks. Tutorial to Gnerative Adversarial Networks and how it's applied using sample code.

"Project Title" should be the "Issue Title" mentioned here.

If you like to work on this issue, follow the given guidelines for code as well as for the README. You have to be as elaborative as possible. Make sure you also maintain the project structure provided. Just comment and get it assigned. Issues are alloted on FCFS basis.

The main intention of this repo is to educate beginners with all the concepts related to datascience so that they could feel confident to apply these things learnt in practical exposure.

Enjoy your open source journey ahead. 😎

**Comment by avinash-218** [comment]
Define You:

- [ ] DevIncept Participant 
- [x] Contributor

**Is your feature request related to a problem? Please describe.**
- I would like to contribute <b>Generative Adversarial Networks</b> using the <a href="https://www.kaggle.com/jessicali9530/celeba-dataset">Celeba Dataset</a>

**Describe the solution you'd like...**

Steps:
- Import Dataset and Libraries
- Setting Parameters and Hyper-parameters
- Config with <a href="https://wandb.ai/home">wandb</a> (optional)
- Define Generator Architecture in Generator Class
- Define Critic Architecture in Critic Class
- Initialize Weights (Optional)
- Define function for gradient penalty calculation
- Define function for saving and loading checkpoints
- A cell for loading previously saved checkpoint
- Training Loop with loggin in wandb
- Cells to generate new faces

**Comment by prathimacode-hub** [comment]
Are you a part of any event or it's a general contribution? @avinash-218

**Comment by avinash-218** [comment]
I'm part of Hacktoberfest 2021

**Comment by prathimacode-hub** [comment]
Ok, issue assigned to @avinash-218
